### PR TITLE
Android media player

### PR DESCRIPTION
**Pull request by theli-ua**
Tested on the emulator

### CODE REVIEW

File: app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java [hunk 1]
```diff
@@ -132,6 +134,21 @@ public static boolean isNetworkAvailable() {
         return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
     }
 
+    public static String getFinalURL(String url) throws IOException {
+        HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+        con.setInstanceFollowRedirects(false);
+        //Pretend we are a Browser, not Android something
+        con.setRequestProperty("User-Agent", "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; en-US; rv:1.9.2.2) Gecko/20100316 Firefox/3.6.2");
```
Comment (mrmaffen) [inline app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java]: I don't think we should hard-code a User-Agent here. Maybe at least make it a parameter?

File: app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                sMediaPlayer = new MediaPlayer();
+
+                sMediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+
+                try {
+                    String finalUrl = NetworkUtils.getFinalURL(path);
+                    if (finalUrl != null)
+                        path = finalUrl;
+                } catch (IOException e) {
+                    e.printStackTrace();
```
Comment (mrmaffen) [inline app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java]: I think calling Log.d(TAG, "prepare - ", e); is better here. mMediaPlayerCallback.onError should also be called here

File: app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                }
+
+                try {
+                    sMediaPlayer.setDataSource(path);
+                    sMediaPlayer.prepare();
+                } catch (IOException e) {
+                    e.printStackTrace();
```
Comment (mrmaffen) [inline app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java]: I think calling Log.d(TAG, "prepare - ", e); is better here.

**Comment by mrmaffen** [comment]
Rest looks fine :+1:

File: app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java (hunk 1 above)
Comment (theli-ua) [inline app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java]: I'd like it to be picked up from resolver's getStreamUrl response, that is ignored currently and I dont think I have access to that here

File: app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java (hunk 2 above)
Comment (theli-ua) [inline app/src/main/java/org/tomahawk/tomahawk_android/mediaplayers/AndroidMediaPlayer.java]: Sure on the Log part. I didn't call onError cause there is still a chance URL will just work with the player, thus no return

File: app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java (hunk 1 above)
Comment (theli-ua) [inline app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java]: Tbh, this part is completely unnecessary for this PR, here native player is used for amazon only and it is unnecessary to get final url for it. I can just remove this from here and see if I can come up with better approach in subsequent PRs, how does that sound?

**Pull request merged**